feat(core): add `optimisticUpdateMap` prop on `useUpdate`, `useUpdateMany` and `useForm` (#4914)
refinedev/refine: documentation/docs/api-reference/core/hooks/data/useUpdate/index.md

console.log(overtime.elapsedTime); // undefined, 1000, 2000, 3000 4000, ...

// You can use it like this:
{
    elapsedTime >= 4000 && <div>this takes a bit longer than expected</div>;
}
```

### `optimisticUpdateMap`

> If the mutation mode is defined as `optimistic` or `undoable` the `useUpdate` hook will automatically update the cache without waiting for the response from the server. You may want to disable or customize this behavior. You can do this by passing the `optimisticUpdateMap` prop.

Reviewer: ```suggestion
If the mutation mode is defined as `optimistic` or `undoable`, `useUpdate` hook will automatically update the cache without waiting for the response from the server. You may want to disable or customize this behavior. You can do this by passing `optimisticUpdateMap` prop.
```

:::caution

> This feature is only work with the `mutationMode` is set to `optimistic` and `undoable`

Reviewer: ```suggestion
This feature only works when `mutationMode` is set to `optimistic` or `undoable`.
```

:::

> `list`, `many` and `detail` are the keys of the `optimisticUpdateMap` object. For automatically updating the cache, you should pass the `true`. If you want not update the cache, you should pass the `false`.

Reviewer: ```suggestion
`list`, `many` and `detail` are the keys of the `optimisticUpdateMap` object. To automatically update the cache, you should pass `true`. If you don't want to update the cache, you should pass `false`.
```

```tsx
const { mutate } = useUpdate();

mutate({
    //...
    mutationMode: "optimistic",
    optimisticUpdateMap: {
        list: true,
        many: true,
        detail: false,
    },
});
```

In the above case the `list` and `many` queries will be updated automatically. But the `detail` query will not be updated.

> Also for customize the cache update, you can pass the function to the `list`, `many` and `detail` keys. The function will be called with the `previous` data, `values` and `id` parameters. You should return the new data from the function.

Reviewer: ```suggestion
Also for customize the cache update, you can pass a function to the `list`, `many` and `detail` keys. These functions will be called with `previous` data, `values` and `id` parameters. You should return  updated data from the function.
```
